### DISCUSSION TITLE
Graph of values directly on the web interface

### DISCUSSION THREAD

**Discussion by rdmueller** [jomjol/AI-on-the-edge-device#1178] [Ideas] [answered]
I just started with this great project to read the values of my gas meter.
What I directly missed was a chart or graph of all values on the web interface.

So I checked how to get the historical data and I found the hint that you can activate debug mode to get the data written to the log. It would be great if the json interface would contain historical data, but the current log also does what I need.

So I then though how I could modify the core software to display the graph. I noticed that the graph can be plottet by the client. So I uploaded a new graph.html file with some javascript to fetch and parse the logs and then to display a graph with an external javascript library. The first PoC work :-)

Is anyone interested in this feature?

**Comment by jomjol** [comment, marked as answer]
Can you publish this part of the code (e.g. in as pull-request in the repository)?

As long as it is "only" html and client based solution, it can be rather easy integrated.

**Reply by rdmueller** [reply]
PR incoming: https://github.com/jomjol/AI-on-the-edge-device/pull/1179
Please keep in mind that this is a first experiment. I am sure I will extend it the next days

**Comment by jomjol** [comment]
Pretty cool! I just saw you commited to master - please use the rolling for updates.

I have added to the rolling as well. Jf you do an update now, it is already part of the html directory.

**Reply by rdmueller** [reply]
Ok. Thanx for the quick merge.
Could you please also mark your repository to be taking part in hacktoberfest.com ?

**Comment by jomjol** [comment]
I don't know, what hacktoberfest is about. Can you explain?

**Reply by rdmueller** [reply]
Hacktoberfest celebrates open source for a whole month. It is sponsored by digital ocean and the goal is to motivate people to contribute to opensource by sending PRs. If you manage to submit 4 PRs, they will plant a tree for you or send you a cool T-Shirt 😁
As a maintainer, you have to tag your repository or contributions to make them count: https://hacktoberfest.com/participation/#maintainers

**Comment by caco3** [comment]
@jomjol Only you can add the label to the repo (gear icon on top right `about` section):
![grafik](https://user-images.githubusercontent.com/1783586/197069865-8e956235-230d-4b3c-8259-8ae6cf093eb3.png)

**Reply by jomjol** [reply]
Why should we participate? I just shortly checked. What would be the benefit?

**Reply by caco3** [reply]
> Why should we participate? I just shortly checked. What would be the benefit?

It gives appreciation to the people sending you pull requests. Kind of an extra motivation to help in Open Source.

**Comment by rdmueller** [comment]
The Benefit is fun, and the chance for more contributions. If your project is tagged, it will be listed on the hacktoberfest.com website. And it is just a tag.

You can also tag single PRs with "hacktoberfest-accepted"

**Comment by forrestfunk** [comment]
Sorry if that's not the right place for this, but I think this is the origin of the "data viewer" in the current build?! 
My question is: is it intented that if theres a "bad" value (like in my case a "rate too high" error-value) that this value gets logged and is shown in the graph?
(I log my values in a influxdb and upon checking the logged values in there, this error-value has not been written to my database but only is shown in my web interface)
(I attached a screenshot of my graph with the false value and my values logged in my influxdb)
sorry again if that's wrong here.
![Screenshot 2022-12-29 165704](https://user-images.githubusercontent.com/10586144/209978450-51e4f7b5-95fd-4711-b463-cd2b8f03ac09.png)
![value iobroker](https://user-images.githubusercontent.com/10586144/209978452-1758a992-5d8f-4462-94b5-69ea43d44ec7.png)

**Reply by caco3** [reply]
Can you please create a bug report for this and also add the part of the logfile?

I agree that it is odd and we should look into it.
